diff --git a/temp_sensor/Core/Src/main.c b/temp_sensor/Core/Src/main.c
index 91b627c..36b37b5 100644
--- a/temp_sensor/Core/Src/main.c
+++ b/temp_sensor/Core/Src/main.c
@@ -21,7 +21,7 @@
 
 /* Private includes ----------------------------------------------------------*/
 /* USER CODE BEGIN Includes */
-
+#include "stdio.h"
 /* USER CODE END Includes */
 
 /* Private typedef -----------------------------------------------------------*/
@@ -45,7 +45,9 @@ ADC_HandleTypeDef hadc1;
 UART_HandleTypeDef huart2;
 
 /* USER CODE BEGIN PV */
-
+const float VOLTAGE_AT_25C = 0.76;
+const float ADC_AVG_SLOPE = 0.0025;
+const float BOARD_REF_VLOTAGE = 3.3;
 /* USER CODE END PV */
 
 /* Private function prototypes -----------------------------------------------*/
@@ -59,7 +61,10 @@ static void MX_USART2_UART_Init(void);
 
 /* Private user code ---------------------------------------------------------*/
 /* USER CODE BEGIN 0 */
-
+int __io_putchar(int ch) {
+  HAL_UART_Transmit(&huart2, (uint8_t *)&ch, 1, HAL_MAX_DELAY);
+  return ch;
+}
 /* USER CODE END 0 */
 
 /**
@@ -101,7 +106,21 @@ int main(void)
   /* USER CODE BEGIN WHILE */
   while (1)
   {
-    /* USER CODE END WHILE */
+    HAL_ADC_Start(&hadc1);
+	HAL_ADC_PollForConversion(&hadc1, HAL_MAX_DELAY);
+
+	uint16_t adc_data = HAL_ADC_GetValue(&hadc1);
+
+	float adc_voltage = ((float)adc_data * BOARD_REF_VLOTAGE) / 4095.0;
+
+	float temperature = ((adc_voltage - VOLTAGE_AT_25C) / ADC_AVG_SLOPE) + 25.0;
+
+    printf("Temperature: %.2f C\n", temperature);
+
+	HAL_ADC_Stop(&hadc1);
+	HAL_Delay(500);
+
+	/* USER CODE END WHILE */
 
     /* USER CODE BEGIN 3 */
   }
