diff --git a/iwdt/Core/Src/main.c b/iwdt/Core/Src/main.c
index 80af4de..b0b0657 100644
--- a/iwdt/Core/Src/main.c
+++ b/iwdt/Core/Src/main.c
@@ -21,7 +21,7 @@
 
 /* Private includes ----------------------------------------------------------*/
 /* USER CODE BEGIN Includes */
-
+#include "stdio.h"
 /* USER CODE END Includes */
 
 /* Private typedef -----------------------------------------------------------*/
@@ -54,6 +54,10 @@ static void MX_GPIO_Init(void);
 static void MX_IWDG_Init(void);
 static void MX_USART2_UART_Init(void);
 /* USER CODE BEGIN PFP */
+int __io_putchar(int ch) {
+  HAL_UART_Transmit(&huart2, (uint8_t *)&ch, 1, HAL_MAX_DELAY);
+  return ch;
+}
 
 /* USER CODE END PFP */
 
@@ -94,6 +98,22 @@ int main(void)
   MX_IWDG_Init();
   MX_USART2_UART_Init();
   /* USER CODE BEGIN 2 */
+  printf("Applications started!\n");
+  printf("IWDT enabled\n");
+
+  printf("Wait 0.5 seconds and feed watchdog\n");
+  HAL_Delay(500);
+  HAL_IWDG_Refresh(&hiwdg);
+
+  printf("Wait another 0.5 seconds and feed watchdog again\n");
+  HAL_Delay(500);
+  HAL_IWDG_Refresh(&hiwdg);
+
+  printf("Wait for reset\n");
+  HAL_Delay(3000);
+
+  printf("Never get here\n");
+
 
   /* USER CODE END 2 */
 
