diff --git a/eeprom_tx_rx_polling/Core/Src/main.c b/eeprom_tx_rx_polling/Core/Src/main.c
index 7dbabb2..3b29a85 100644
--- a/eeprom_tx_rx_polling/Core/Src/main.c
+++ b/eeprom_tx_rx_polling/Core/Src/main.c
@@ -18,6 +18,7 @@
 /* USER CODE END Header */
 /* Includes ------------------------------------------------------------------*/
 #include "main.h"
+#include "stdio.h"
 
 /* Private includes ----------------------------------------------------------*/
 /* USER CODE BEGIN Includes */
@@ -31,7 +32,8 @@
 
 /* Private define ------------------------------------------------------------*/
 /* USER CODE BEGIN PD */
-
+#define EEPROM_ADDR 0xA0
+#define COUNTER_ADDR 0x00
 /* USER CODE END PD */
 
 /* Private macro -------------------------------------------------------------*/
@@ -54,7 +56,10 @@ static void MX_GPIO_Init(void);
 static void MX_I2C1_Init(void);
 static void MX_USART2_UART_Init(void);
 /* USER CODE BEGIN PFP */
-
+int __io_putchar(int ch) {
+  HAL_UART_Transmit(&huart2, (uint8_t *)&ch, 1, HAL_MAX_DELAY);
+  return ch;
+}
 /* USER CODE END PFP */
 
 /* Private user code ---------------------------------------------------------*/
@@ -94,13 +98,29 @@ int main(void)
   MX_I2C1_Init();
   MX_USART2_UART_Init();
   /* USER CODE BEGIN 2 */
-
+  uint8_t counter;
+  uint8_t tx_data[2] = {COUNTER_ADDR, 0};
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+    // set address inside EEPROM
+	HAL_I2C_IsDeviceReady(&hi2c1, EEPROM_ADDR, 1, HAL_MAX_DELAY);
+    HAL_I2C_Master_Transmit(&hi2c1, EEPROM_ADDR, tx_data, 1, HAL_MAX_DELAY);
+
+    // read counter value
+	HAL_I2C_IsDeviceReady(&hi2c1, EEPROM_ADDR, 1, HAL_MAX_DELAY);
+	HAL_I2C_Master_Receive(&hi2c1, EEPROM_ADDR, &counter, 1, HAL_MAX_DELAY);
+    printf("Counter value from EEPROM: %d\n", (int)counter);
+
+    // incremetn counter and write it back
+    tx_data[1] = ++counter;
+    HAL_I2C_Master_Transmit(&hi2c1, EEPROM_ADDR, tx_data, 2, HAL_MAX_DELAY);
+
+    // wait 1 second and start again
+    HAL_Delay(1000);
     /* USER CODE END WHILE */
 
     /* USER CODE BEGIN 3 */
