diff --git a/blink/Core/Src/main.c b/blink/Core/Src/main.c
index b4fb5a1..cc8669e 100644
--- a/blink/Core/Src/main.c
+++ b/blink/Core/Src/main.c
@@ -42,10 +42,9 @@
 
 /* Private variables ---------------------------------------------------------*/
 /* Definitions for defaultTask */
-osThreadId_t defaultTaskHandle;
-const osThreadAttr_t defaultTask_attributes = {
-  .name = "defaultTask",
-  .stack_size = 128 * 4,
+const osThreadAttr_t thread_attributes = {
+  .name = "blink_task",
+  .stack_size = 1024 * 4,
   .priority = (osPriority_t) osPriorityNormal,
 };
 /* USER CODE BEGIN PV */
@@ -55,7 +54,7 @@ const osThreadAttr_t defaultTask_attributes = {
 /* Private function prototypes -----------------------------------------------*/
 void SystemClock_Config(void);
 static void MX_GPIO_Init(void);
-void StartDefaultTask(void *argument);
+void blink_thread(void *argument);
 
 /* USER CODE BEGIN PFP */
 
@@ -120,7 +119,7 @@ int main(void)
 
   /* Create the thread(s) */
   /* creation of defaultTask */
-  defaultTaskHandle = osThreadNew(StartDefaultTask, NULL, &defaultTask_attributes);
+  osThreadNew(blink_thread, NULL, &thread_attributes);
 
   /* USER CODE BEGIN RTOS_THREADS */
   /* add threads, ... */
@@ -228,13 +227,14 @@ static void MX_GPIO_Init(void)
   * @retval None
   */
 /* USER CODE END Header_StartDefaultTask */
-void StartDefaultTask(void *argument)
+void blink_thread(void *argument)
 {
   /* USER CODE BEGIN 5 */
   /* Infinite loop */
   for(;;)
   {
-    osDelay(1);
+	HAL_GPIO_TogglePin(LED_GPIO_Port, LED_Pin);
+    osDelay(500);
   }
   /* USER CODE END 5 */
 }
