diff --git a/base_dma/Core/Src/main.c b/base_dma/Core/Src/main.c
index 4b992e5..cb0ca82 100644
--- a/base_dma/Core/Src/main.c
+++ b/base_dma/Core/Src/main.c
@@ -58,7 +58,19 @@ static void MX_TIM1_Init(void);
 
 /* Private user code ---------------------------------------------------------*/
 /* USER CODE BEGIN 0 */
+void HAL_TIM_PeriodElapsedCallback(TIM_HandleTypeDef *htim)
+{
+  if(htim == &htim1) {
+	HAL_GPIO_TogglePin(LED_GPIO_Port, LED_Pin);
+  }
+}
 
+void HAL_TIM_PeriodElapsedHalfCpltCallback(TIM_HandleTypeDef *htim)
+{
+  if(htim == &htim1) {
+	HAL_GPIO_TogglePin(LED_GPIO_Port, LED_Pin);
+  }
+}
 /* USER CODE END 0 */
 
 /**
@@ -98,6 +110,13 @@ int main(void)
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+  uint16_t periods[2] = {
+    200  - 1,
+	1000  - 1
+  };
+
+  HAL_TIM_Base_Start_DMA(&htim1, (const uint32_t*)periods, 2);
+
   while (1)
   {
     /* USER CODE END WHILE */
