diff --git a/button_exti/Core/Src/main.c b/button_exti/Core/Src/main.c
index aabf9b1..b728198 100644
--- a/button_exti/Core/Src/main.c
+++ b/button_exti/Core/Src/main.c
@@ -183,7 +183,12 @@ static void MX_GPIO_Init(void)
 }
 
 /* USER CODE BEGIN 4 */
-
+void HAL_GPIO_EXTI_Callback(uint16_t GPIO_Pin)
+{
+  if (GPIO_Pin == BUTTON_Pin) {
+	  HAL_GPIO_TogglePin(LED_GPIO_Port, LED_Pin);
+  }
+}
 /* USER CODE END 4 */
 
 /**
