diff --git a/pwm_led/Core/Src/main.c b/pwm_led/Core/Src/main.c
index 2a613b8..8f74e82 100644
--- a/pwm_led/Core/Src/main.c
+++ b/pwm_led/Core/Src/main.c
@@ -90,13 +90,22 @@ int main(void)
   MX_GPIO_Init();
   MX_TIM2_Init();
   /* USER CODE BEGIN 2 */
-
+  HAL_TIM_PWM_Start(&htim2, TIM_CHANNEL_1);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+  uint32_t pulse_width = 0;
   while (1)
   {
+	__HAL_TIM_SET_COMPARE(&htim2, TIM_CHANNEL_1, pulse_width);
+	pulse_width++;
+
+	if (pulse_width >= 100) {
+		pulse_width = 0;
+	}
+
+	HAL_Delay(10);
     /* USER CODE END WHILE */
 
     /* USER CODE BEGIN 3 */
